src/serializer/ResidualFunctions.js

```diff
@@ -54,6 +54,7 @@ export class ResidualFunctions {
     initializerNameGenerator: NameGenerator,
     factoryNameGenerator: NameGenerator,
     scopeNameGenerator: NameGenerator,
+    capturedScopeGenerator: NameGenerator,
```

Comment:
Let's keep it simple. We don't really need a new name generator instance. Remove it.